### PR TITLE
96 update terraform for openApi

### PR DESCRIPTION
**Pull request by mahinuzzaman**
Feat: #96 
- Added terraform and scripts for preparing and importing openApi to ApiGateway.
- Removed api_gw module from Terraform. cause the API gateway integration will be defined in openApi.
- Updated docker-compose file for merged openApi file to run swagger locally
- also, for now, added an example code for auth/register-org path (which has a separate ticket #75)

for openApi files please look at the previous PR #97 

- [ ] Does not depend on other PRs
- [x] Depends on other PR being merged first: #97

## What kind of change does this PR introduce (check at least one)

- [ ] Bugfix
- [x] Improvement
- [x] New feature
- [ ] Refactor
- [ ] Infrastructure
- [ ] Other (please specify)

## How would you describe the changes in a changelog (1-2 lines)?

Provide a brief overview of the changes introduced in this pull request.

## Related Issues/PRs (optional)

Link to the relevant issue(s) or provide a clear description of the problem or feature being addressed.

## How to test

- [ ] Needs regression testing (add QA needed label)

## Checklist

- [ ] I have tested these changes locally and they function as expected.
- [ ] I have added or updated necessary documentation to reflect the changes.
- [ ] I have followed the project's coding style and guidelines.
- [ ] I have added appropriate test coverage to ensure the changes are functioning correctly.
- [ ] I have checked for any potential conflicts with other branches or pull requests.
- [ ] I have updated necessary OpenAPI specs

## Reviewer Request (optional)

If you have any specific requests or suggestions for reviewers, mention them here.

## Screenshots (optional)

If applicable, provide any additional screenshots or GIFs that showcase the changes made in this pull request.

## Demo Link (optional)

If applicable, provide a link to a live demo or deployment where the changes can be reviewed.

## Known Issues (optional)

If there are any known issues or limitations with the changes made in this pull request, describe them here.

## Additional Notes (optional)

Add any additional notes or comments you think are relevant to this pull request.

## Deploy risk & follow up

Please assess risk & follow up on deployment of this PR. Elaborate your choice if selecting anything other than 'No risk'.

- [x] No risk
- [ ] Low
- [ ] High

### CODE REVIEW

File: iac/terraform/aws/scripts.tf [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    environment = {
+      OPENAPI_DIRECTORY = var.openapi_files_path
+      API_VERSION       = var.openapi_version
+      CLOUD_PROVIDER    = "aws"
+    }
+    command = "${path.module}/scripts/updateOpenApiIntegrations.sh"
+  }
+
+  depends_on = [
+    aws_api_gateway_rest_api.rest_api
```
Comment (melon-ruet) [inline iac/terraform/aws/scripts.tf:12]: Does it depend on other scripts?

File: iac/terraform/aws/scripts/prepOpenApi.sh
Comment (melon-ruet) [inline iac/terraform/aws/scripts/prepOpenApi.sh:1]: better name `prepareOpenApi.sh`

**Review by melon-ruet — CHANGES_REQUESTED** [review]

File: iac/terraform/aws/scripts.tf (hunk 1 above)
Comment (mahinuzzaman) [reply iac/terraform/aws/scripts.tf:12]: no, this is the first script that runs after API gateway creation

File: iac/terraform/aws/scripts/prepOpenApi.sh
Comment (mahinuzzaman) [reply iac/terraform/aws/scripts/prepOpenApi.sh:1]: Moved all import api scripts in to a single script

File: iac/terraform/aws/auth/lambdas.tf [hunk 2]
```diff
@@ -8,6 +8,13 @@ locals {
       env_variables = {
         foo = "bar"
       }
+    },
+    register-org = {
```
Comment (melon-ruet) [inline iac/terraform/aws/auth/lambdas.tf:12]: register-organization

File: iac/terraform/aws/auth/lambdas.tf [hunk 3]
```diff
@@ -8,6 +8,13 @@ locals {
...
+      name          = "register-org"
```
Comment (melon-ruet) [inline iac/terraform/aws/auth/lambdas.tf:13]: register-organization

**Review by melon-ruet — CHANGES_REQUESTED** [review]

**Review by melon-ruet — APPROVED** [review]

**Pull request merged**